src/entities/raw-data-providers/cardano-explorer-api/cardano-explorer-api.js

```diff
@@ -0,0 +1,175 @@
+import { ByronBlock } from '../../../blockchain/byron'
+
+import {
+  sql,
+  GET_BEST_BLOCK_NUM,
+} from './db-queries'
+
+class CardanoExplorerApi implements RawDataProvider {
+  logger: Logger
+
+  conn: DBConnection
+
+  networkConfig: NetworkConfig
+
+  constructor(
+    networkConfig: NetworkConfig,
+    logger: Logger,
+    conn: DBConnection,
+  ) {
+    this.logger = logger
+    this.networkConfig = networkConfig
+    this.conn = conn
+  }
+
+  async getBlock(id: string): Promise<ByronBlock> {
+    this.logger.debug(`[cardano-explorer] GET BLOCK: ${id}`)
+    const blockSql = sql.select()
+      .from('"Block"', 'block')
+      .fields({
+        'block.number': 'number',
+        'block.id': 'id',
+        'block."epochNo"': '"epochNo"',
+        'block."slotNo"': '"slotNo"',
+        'block.size': 'size',
+        'block."createdAt"': '"createdAt"',
+        'block."createdBy"': '"createdBy"',
+        'slot_leader.hash': '"leaderHash"',
+        'block."previousBlockId"': '"previousBlockId"',
+      })
+      .left_join('slot_leader', null, 'block."createdBy" = slot_leader.description')
+      .where('block.id = ?', `\\x${id}`)
+      .limit(1)
+      .toString()
+    const txs = []
+    const blockData = (await this.conn.query(blockSql)).rows[0]
+    const blockId = blockData.id.toString('hex')
+    this.logger.debug('blockData', blockData)
+
+    const txsSql = sql.select().from('"Transaction"')
+      .where('"blockId" = ?', `\\x${blockId}`).toString()
+    const txsData = (await this.conn.query(txsSql)).rows
+    if (txsData.length > 0 && blockData.number !== undefined) {
+      const txsIds = txsData.map(tx => `\\x${tx.id.toString('hex')}`)
+      // get transaction inputs
+      const txInputsSql = sql.select().from('"TransactionInput"')
+        .where('"txId" IN ?', txsIds).toString()
+      const txInputsData = (await this.conn.query(txInputsSql)).rows
+      // get transaction outputs
+      const txOutputsSql = sql.select().from('"TransactionOutput"')
+        .where('"txId" IN ?', txsIds).toString()
+      const txOutputsData = (await this.conn.query(txOutputsSql)).rows
+      for (const txData of txsData) {
+        txs.push({
+          inputs: txInputsData
+            .filter((inp) => inp.txId.toString('hex') === txData.id.toString('hex'))
+            .map((inp) => ({
+              type: 'utxo',
+              txId: inp.sourceTxId.toString('hex'),
+              idx: inp.sourceTxIndex,
+            })),
+          outputs: txOutputsData
+            .filter((out) => out.txId.toString('hex') === txData.id.toString('hex'))
+            .map((out) => ({
+              type: 'utxo',
+              address: out.address,
+              value: out.value,
+            })),
+          id: txData.id.toString('hex'),
+          blockNum: blockData.number,
+          blockHash: blockData.id.toString('hex'),
+          status: 'Successful',
+          txTime: txData.includedAt,
+          txBody: '',
+          txOrdinal: 0,
```

Comment:
I understand that `txBody` is empty because we don't have this in cardano-explorer (but probably there should be a comment for this). I don't understand why `txOrdinal` is always 0 though